File: chapter-02/exercise-06-retirement-calculator/RetirementCalculator.java
import java.time.Year;

class RetirementCalculator {
  public static void main(String[] args) {
    var input = System.console().readLine("What is your current age? ");
    var currentAge = Integer.parseInt(input);

    input = System.console().readLine("At what age would you like to retire? ");
    var retireAge = Integer.parseInt(input);

    var yearsLeft = retireAge - currentAge;
    var currentYear = Year.now().getValue();
    var retirementYear = currentYear + yearsLeft;

    if (yearsLeft < 1) {
      System.out.println("You can already retire.");
    } else {
      System.out.printf("You have %d years left until you can retire.%nIt's %d, so you can retire in %d.%n",
        yearsLeft, currentYear, retirementYear);
    }
  }
}

File: 06-retirement-calculator/retirement-calculator.js
const readline = require('readline');
const rl = readline.createInterface(process.stdin, process.stdout);

rl.question('What is your current age? ', (answer) => {
  const currentAge = parseInt(answer);
  rl.question('At what age would you like to retire? ', (answer) => {
    const retireAge = parseInt(answer);

    const yearsLeft = retireAge - currentAge;
    const currentYear = new Date().getFullYear();
    const retireYear = currentYear + yearsLeft;

    if (yearsLeft < 1) {
      console.log(`You can already retire!`);
    } else {
      console.log(`You have ${yearsLeft} years left until you can retire.\nIt's ${currentYear}, so you can retire in ${retireYear}.`);
    }

    rl.close();
  });
});